@here/olp-sdk-core/lib/utils/TileKey.ts

```diff
@@ -0,0 +1,421 @@
+     * @param level The requested level.
+     */
+    static fromRowColumnLevel(
+        row: number,
+        column: number,
+        level: number
+    ): TileKey {
+        return new TileKey(row, column, level);
+    }
+
+    /**
+     * Creates a tile key from a quad string.
+     *
+     * The quad string can be created with [[toQuadKey]].
+     *
+     * @param quadkey The quadkey to convert.
+     * @returns A new instance of `TileKey`.
+     */
+    static fromQuadKeyString(key: string): TileKey {
+        if (key === "-") {
+            return TileKey.fromRowColumnLevel(0, 0, 0);
+        }
+
+        const level = key.length;
+        let row = 0;
+        let column = 0;
+        // tslint:disable:no-bitwise
+        for (let i = 0; i < key.length; ++i) {
+            const mask = 1 << i;
+            const d = parseInt(key.charAt(level - i - 1), 10);
+            if (d & 0x1) {
+                column |= mask;
+            }
+            if (d & 0x2) {
+                row |= mask;
+            }
+        }
+        // tslint:enable:no-bitwise
+        return TileKey.fromRowColumnLevel(row, column, level);
+    }
+
+    /**
+     * Creates a tile key from a numeric Morton code representation.
+     *
+     * You can convert a tile key into a numeric Morton code with [[mortonCode]].
+     *
+     * @param key The Morton code to be converted.
+     * @returns A new instance of [[TileKey]].
+     */
+    static fromMortonCode(key: number): TileKey {
+        let level = 0;
+        let row = 0;
+        let column = 0;
+        let quadKey = key;
+        // tslint:disable:no-bitwise
+        while (quadKey > 1) {
+            const mask: number = 1 << level;
+
+            if (quadKey & 0x1) {
+                column |= mask;
+            }
+            if (quadKey & 0x2) {
+                row |= mask;
+            }
+
+            level++;
+            // tslint:disable-next-line: no-magic-numbers
+            quadKey = (quadKey - (quadKey & 0x3)) / SUBTILES_COUNT;
+        }
+        // tslint:enable:no-bitwise
+        const result = TileKey.fromRowColumnLevel(row, column, level);
+        result.m_mortonCode = key;
+        return result;
+    }
+
+    /**
+     * Returns the number of available columns at a given level.
+     *
+     * This is 2 to the power of the level.
+     *
+     * @param level The level for which to return the number of columns.
+     * @returns The available columns at the given level.
+     */
+    static columnsCount(level: number): number {
+        return Math.pow(2, level);
+    }
+
+    /**
+     * Returns the number of available rows at a given level.
+     *
+     * This is 2 to the power of the level.
+     *
+     * @param level The level for which to return the number of rows.
+     * @returns The available rows at the given level.
+     */
+    static rowsCount(level: number): number {
+        return Math.pow(2, level);
+    }
```

Comment:
You can do a simple trick here by using shift which basically gives you power of 2
```ts
return 1 << level;
```